controllers/backup-service/reconciler.go

```diff
@@ -0,0 +1,684 @@
+		r.Log.Info("HTTP config not found")
+		return &defaultServiceConfig, nil
+	}
+
+	httpConf, ok := svc[common.HTTPKey].(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("http config is not in correct format")
+	}
+
+	var svcConfig serviceConfig
+
+	port, ok := httpConf["port"]
+	if !ok {
+		svcConfig.portInfo = defaultServiceConfig.portInfo
+	} else {
+		svcConfig.portInfo = map[string]int32{common.HTTPKey: int32(port.(float64))}
+	}
+
+	ctxPath, ok := httpConf["context-path"]
+	if !ok {
+		svcConfig.contextPath = defaultServiceConfig.contextPath
+	} else {
+		svcConfig.contextPath = ctxPath.(string)
+	}
+
+	return &svcConfig, nil
+}
+
+func (r *SingleBackupServiceReconciler) waitForDeploymentToBeReady() error {
+	const (
+		podStatusTimeout       = 2 * time.Minute
+		podStatusRetryInterval = 5 * time.Second
+	)
+
+	r.Log.Info(
+		"Waiting for deployment to be ready", "WaitTimePerPod", podStatusTimeout,
+	)
+
+	if err := wait.PollUntilContextTimeout(context.TODO(),
+		podStatusRetryInterval, podStatusTimeout, true, func(ctx context.Context) (done bool, err error) {
+			podList, err := r.getBackupServicePodList()
+			if err != nil {
+				return false, err
+			}
+
+			if len(podList.Items) == 0 {
+				r.Log.Info("No pod found for deployment")
+				return false, nil
+			}
+
+			for idx := range podList.Items {
+				pod := &podList.Items[idx]
+
+				if err := utils.CheckPodFailed(pod); err != nil {
+					return false, fmt.Errorf("pod %s failed: %v", pod.Name, err)
+				}
+
+				if !utils.IsPodRunningAndReady(pod) {
+					r.Log.Info("Pod is not ready", "pod", pod.Name)
+					return false, nil
+				}
+			}
+
+			var deploy app.Deployment
+			if err := r.Client.Get(
+				ctx,
+				types.NamespacedName{Name: r.aeroBackupService.Name, Namespace: r.aeroBackupService.Namespace},
+				&deploy,
+			); err != nil {
+				return false, err
+			}
+
+			if deploy.Status.Replicas != *deploy.Spec.Replicas {
+				return false, nil
+			}
+
+			return true, nil
+		},
+	); err != nil {
+		return err
+	}
+
+	r.Log.Info("Deployment is ready")
+
+	return nil
+}
+
+func (r *SingleBackupServiceReconciler) setStatusPhase(phase asdbv1beta1.AerospikeBackupServicePhase) error {
+	if r.aeroBackupService.Status.Phase != phase {
+		r.aeroBackupService.Status.Phase = phase
+
+		if err := r.Client.Status().Update(context.Background(), r.aeroBackupService); err != nil {
+			r.Log.Error(err, fmt.Sprintf("Failed to set backup service status to %s", phase))
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *SingleBackupServiceReconciler) updateStatus() error {
+	svcConfig, err := r.getBackupServiceConfig()
+	if err != nil {
+		return err
+	}
+
+	status := r.CopySpecToStatus()
+	status.ContextPath = svcConfig.contextPath
+	status.Port = svcConfig.portInfo[common.HTTPKey]
+	status.Phase = asdbv1beta1.AerospikeBackupServiceCompleted
+
+	r.aeroBackupService.Status = *status
+
+	r.Log.Info(fmt.Sprintf("Updating status: %+v", r.aeroBackupService.Status))
```

Comment:
Raw data printed here is unreadable, either skip it or print in readable format.
Current output is: `Updating status: {Image:aerospike.jfrog.io/ecosystem-container-prod-local/aerospike-backup-service:1.0.0 Config:{Raw:[123 34 98 97 99 107 117 112 45 112 111.....`

Reply:
Removed